tests/mapper/test_graphwave_generator.py

```diff
@@ -0,0 +1,272 @@
+
+        for i, x in enumerate(
+            generator.flow(
+                self.G.nodes(),
+                sample_points=self.sample_points,
+                batch_size=batch_size,
+                repeat=False,
+            )
+        ):
+            # all batches except maybe last will have a batch size of batch_size
+            if i < self.num_nodes // batch_size:
+                assert x.shape == (batch_size, expected_embed_dim)
+            else:
+                assert x.shape == (self.num_nodes % batch_size, expected_embed_dim)
+
+    @pytest.mark.parametrize("num_samples", [1, 25, 50])
+    def test_embedding_dim(self, num_samples):
+        scales = (0.1, 2, 3, 4)
+        generator = GraphWaveGenerator(self.G, scales=scales, deg=10)
+
+        sample_points = np.linspace(0, 1, num_samples)
+
+        expected_embed_dim = len(sample_points) * len(scales) * 2
+
+        for x in generator.flow(
+            self.G.nodes(), sample_points=sample_points, batch_size=4, repeat=False
+        ):
+
+            assert x.shape[1] == expected_embed_dim
+
+    def test_flow_targets(self):
+
+        generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=10)
+
+        for i, x in enumerate(
+            generator.flow(
+                self.G.nodes(),
+                sample_points=self.sample_points,
+                batch_size=1,
+                targets=np.arange(self.num_nodes),
+            )
+        ):
+            assert len(x) == 2
+            assert x[1].numpy() == i
+
+    def test_flow_node_ids(self):
+
+        generator = GraphWaveGenerator(self.G, scales=(0.1, 2, 3, 4), deg=10)
+
+        node_ids = list(self.G.nodes())[:4]
+
+        expected_targets = generator._node_lookup(node_ids)
+        actual_targets = []
+        for x in generator.flow(
+            node_ids,
+            sample_points=self.sample_points,
+            batch_size=1,
+            targets=expected_targets,
+        ):
+
+            actual_targets.append(x[1].numpy())
+
+        assert all(a == b for a, b in zip(expected_targets, actual_targets))
+
+    def test_chebyshev(self):
+        """
+        This test checks that the Chebyshev approximation accurately calculates the wavelets. It calculates
+        the wavelets exactly using eigenvalues and compares this to the Chebyshev approximation.
+        """
+        scales = (1, 5, 10)
+
+        generator = GraphWaveGenerator(self.G, scales=scales, deg=50,)
+
+        # calculate wavelets exactly using eigenvalues
+        adj = self.G.to_adjacency_matrix().tocoo()
+
+        degree_mat = sps.diags(np.asarray(adj.sum(1)).ravel())
+        laplacian = degree_mat - adj
+        laplacian = np.asarray(laplacian.todense()).astype(np.float64)
+
+        eigenvals, eigenvecs = np.linalg.eig(laplacian)
+        eigenvecs = np.asarray(eigenvecs)
+
+        psis = [
+            (eigenvecs * np.exp(-s * eigenvals)).dot(eigenvecs.transpose())
```

Comment:
Since this is designed to be a test, I'd be ok with doing this with the "textbook definition" way U g_s(Λ) U^T, so that it's more "obviously" correct:

```suggestion
            eigenvecs @ np.diag(np.exp(-s * eigenvals))) @ eigenvecs.T
```